### PR TITLE
pkg/gitinterface: Add tests for uncovered error paths and edge cases

### PR DESCRIPTION
**Pull request by dawgdevv**
## Description

This change improves test coverage for pkg/gitinterface by adding tests for previously uncovered error paths and edge cases:

- Tree operations: ensureIsTree non-existent object, GetPathIDInTree and GetTreeItems with blob/non-existent IDs, GetMergeTree with non-commit objects, GetAllFilesInTree errors, NewEntryBlobWithPermissions, NewEntryTree with nil/ZeroHash/valid ID

- Signature operations: CanSign with SSH format but no signing key, VerifySignature with non-commit/tag object, signGitObjectUsingKey with unknown PEM block type, invalid GPG and SSH key errors

- Commit operations: CommitUsingSpecificKey with invalid signing key

- Tag operations: ensureIsTag with non-existent object, TagUsingSpecificKey with invalid target

- Reference operations: CheckAndSetReference with wrong old ID, SetReference/DeleteReference with invalid ref names, RefSpec with non-existent ref

- Repository operations: LoadRepository with empty and invalid paths

- Changes operations: GetFilePathsChangedByCommit with non-commit object

Coverage: 76.0% -> 78.3%
## AI Usage

- [ ] I **did not** use generative AI at all in making the content of this pull
  request.
- [x] I **did** use generative AI in some form in making the content of this
  pull request. I have described my use of AI below.

This PR was prepared with the assistance of an AI coding assistant. All changes were reviewed manually for correctness, style compliance, and alignment with project conventions.

## Contributor Checklist

- [x] I **have manually reviewed all content** submitted to gittuf in this pull
  request.
- [x] I fully understand the content I am submitting.
- [x] The changes introduced are documented and have tests included if
  applicable.
- [x] My changes do not infringe on copyright/trademarks/etc.
- [x] All commits in this pull request include a [DCO
  Signoff](https://wiki.linuxfoundation.org/dco).
- [x] By submitting this pull request, I agree to follow the gittuf [Code of
  Conduct](https://github.com/gittuf/community/blob/main/CODE-OF-CONDUCT.md).

### CODE REVIEW

**Comment by dawgdevv** [comment]
@patzielinski added some more tests take a look if something needs to be addressed

File: pkg/gitinterface/commit_test.go [hunk 1]
```diff
@@ -131,6 +131,22 @@ func TestRepositoryCommitUsingSpecificKey(t *testing.T) {
...
+
+	refName := "refs/heads/main"
+	treeBuilder := NewTreeBuilder(repo)
+
+	emptyTreeID, err := treeBuilder.WriteTreeFromEntries(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = repo.CommitUsingSpecificKey(emptyTreeID, refName, "Initial commit\n", []byte("-----BEGIN UNKNOWN KEY-----\nYWJj\n-----END UNKNOWN KEY-----\n"))
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go]: Can the key be moved out to a variable above?

File: pkg/gitinterface/commit_test.go [hunk 2]
```diff
@@ -131,6 +131,22 @@ func TestRepositoryCommitUsingSpecificKey(t *testing.T) {
 	assert.Equal(t, expectedSecondCommitID, refHead.String())
 }
 
+func TestRepositoryCommitUsingSpecificKeyInvalidKey(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go]: Can this be inlined with `TestRepositoryCommitUsingSpecificKey`?

File: pkg/gitinterface/references_test.go [hunk 3]
```diff
@@ -461,3 +461,60 @@ func TestAbsoluteReference(t *testing.T) {
 		assert.ErrorIs(t, err, ErrReferenceNotFound)
 	})
 }
+
+func TestCheckAndSetReferenceError(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/references_test.go]: Please inline this with `TestCheckAndSetReference`.

File: pkg/gitinterface/references_test.go [hunk 4]
```diff
@@ -461,3 +461,60 @@ func TestAbsoluteReference(t *testing.T) {
...
+	require.Nil(t, err)
+
+	_, err = repo.Commit(emptyTreeID, refName, "Second commit\n", false)
+	require.Nil(t, err)
+
+	err = repo.CheckAndSetReference(refName, firstCommitID, firstCommitID)
+	assert.ErrorContains(t, err, "unable to set Git reference")
+}
+
+func TestSetReferenceError(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/references_test.go]: Please inline this with `TestSetReference`.

File: pkg/gitinterface/references_test.go [hunk 5]
```diff
@@ -461,3 +461,60 @@ func TestAbsoluteReference(t *testing.T) {
...
+	}
+
+	commitID, err := repo.Commit(emptyTreeID, refName, "Initial commit\n", false)
+	require.Nil(t, err)
+
+	err = repo.SetReference("invalid ref name", commitID)
+	assert.ErrorContains(t, err, "unable to set Git reference")
+}
+
+func TestDeleteReferenceError(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/references_test.go]: Please inline this with `TestDeleteReference`.

File: pkg/gitinterface/references_test.go [hunk 6]
```diff
@@ -461,3 +461,60 @@ func TestAbsoluteReference(t *testing.T) {
...
+	tmpDir := t.TempDir()
+	repo := CreateTestGitRepository(t, tmpDir, false)
+
+	err := repo.DeleteReference("invalid ref name")
+	assert.ErrorContains(t, err, "unable to delete Git reference")
+}
+
+func TestRepositoryRefSpecError(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/references_test.go]: Please inline this with `TestRepositoryRefSpec`.

File: pkg/gitinterface/repository_test.go [hunk 7]
```diff
@@ -53,4 +54,19 @@ func TestRepository(t *testing.T) {
...
 	})
+
+	t.Run("empty path", func(t *testing.T) {
+		_, err := LoadRepository("")
+		assert.ErrorIs(t, err, ErrRepositoryPathNotSpecified)
+	})
+
+	t.Run("invalid path", func(t *testing.T) {
+		tmpDir := t.TempDir()
+		notARepoDir := filepath.Join(tmpDir, "not-a-repo")
```
Comment (patzielinski) [inline pkg/gitinterface/repository_test.go]: Could this be simplified to just pass in the temp dir?

File: pkg/gitinterface/signature_test.go [hunk 8]
```diff
@@ -73,3 +75,41 @@ func TestCanSign(t *testing.T) {
 		})
 	}
 }
+
+func TestCanSignSSHNoKey(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/signature_test.go]: Please inline this with `TestCanSign`.

File: pkg/gitinterface/signature_test.go [hunk 9]
```diff
@@ -73,3 +75,41 @@ func TestCanSign(t *testing.T) {
...
+
+	if err := repo.SetGitConfig("gpg.format", "ssh"); err != nil {
+		t.Fatal(err)
+	}
+
+	err := repo.CanSign()
+	assert.ErrorIs(t, err, ErrSigningKeyNotSpecified)
+}
+
+func TestVerifySignatureNotCommitOrTag(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/signature_test.go]:
```suggestion
func TestVerifySignature(t *testing.T) {
     t.Run("not a commit or a tag", func(t *testing.T) {
```

File: pkg/gitinterface/signature_test.go [hunk 10]
```diff
@@ -73,3 +75,41 @@ func TestCanSign(t *testing.T) {
...
+
+func TestSignGitObjectUsingGPGKeyInvalid(t *testing.T) {
+	_, err := signGitObjectUsingGPGKey([]byte("test"), []byte("invalid"))
+	assert.ErrorContains(t, err, "openpgp")
+}
+
+func TestSignGitObjectUsingSSHKeyInvalid(t *testing.T) {
+	_, err := signGitObjectUsingSSHKey([]byte("test"), []byte("invalid"))
+	assert.ErrorContains(t, err, "ssh")
+}
```
Comment (patzielinski) [inline pkg/gitinterface/signature_test.go]: I think we can just omit these tests.

File: pkg/gitinterface/tag_test.go [hunk 11]
```diff
@@ -126,4 +126,15 @@ func TestEnsureIsTag(t *testing.T) {
 
 	err = repo.ensureIsTag(commitID)
 	assert.ErrorContains(t, err, "is not a tag object")
+
+	err = repo.ensureIsTag(ZeroHash)
+	assert.ErrorContains(t, err, "unable to inspect if object is tag")
+}
+
+func TestTagUsingSpecificKeyInvalidTarget(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/tag_test.go]:
```suggestion
func TestTagUsingSpecificKey(t *testing.T) {
```

Comment (patzielinski) [reply pkg/gitinterface/tag_test.go]: Same as before, please put this in a `t.Run`.

File: pkg/gitinterface/tree_test.go [hunk 12]
```diff
@@ -1121,4 +1178,61 @@ func TestEnsureIsTree(t *testing.T) {
 
 	err = repo.ensureIsTree(blobID)
 	assert.NotNil(t, err)
+
+	err = repo.ensureIsTree(ZeroHash)
+	assert.ErrorContains(t, err, "unable to inspect if object is tree")
+}
+
+func TestGetAllFilesInTreeError(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/tree_test.go]:
```suggestion
func TestGetAllFilesInTree(t *testing.T) {
```
Same as before, please put the test inside in an appropriately-named `t.Run`.

File: pkg/gitinterface/tree_test.go [hunk 13]
```diff
@@ -1121,4 +1178,61 @@ func TestEnsureIsTree(t *testing.T) {
...
+	}
+
+	_, err = repo.GetAllFilesInTree(blobID)
+	assert.ErrorContains(t, err, "unable to enumerate all files in tree")
+
+	_, err = repo.GetAllFilesInTree(ZeroHash)
+	assert.ErrorContains(t, err, "unable to enumerate all files in tree")
+}
+
+func TestNewEntryBlobWithPermissions(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/tree_test.go]: I don't think we use this functionality at the moment, so we can omit this test.

File: pkg/gitinterface/tree_test.go [hunk 14]
```diff
@@ -1121,4 +1178,61 @@ func TestEnsureIsTree(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	entry := NewEntryBlobWithPermissions("a", blobID, 0o755)
+	assert.Equal(t, "a", entry.getName())
+	assert.Equal(t, blobID, entry.getID())
+}
+
+func TestNewEntryTree(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/tree_test.go]: I think this is tested sufficiently by the other functions, so we can omit it.

**Review by patzielinski — CHANGES_REQUESTED** [review]
Thanks for sending this in, a few nits.

File: pkg/gitinterface/commit_test.go (hunk 2 above)
Comment (dawgdevv) [reply pkg/gitinterface/commit_test.go]: i think we should keep this separate TestRepositoryCommitUsingSpecificKey already has two hard coded  hashes a separate function almost costs nothing to keep this  if you really want it inlined we can do it no problem in that

**Comment by dawgdevv** [comment]
@patzielinski I have made the changes suggest if something still need improvement

File: pkg/gitinterface/commit_test.go (hunk 2 above)
Comment (patzielinski) [reply pkg/gitinterface/commit_test.go]: I know, but this is about keeping the testing format uniform; let's just inline this and we can reorganize some tests to make them cleaner at a later date.

Comment (dawgdevv) [reply pkg/gitinterface/commit_test.go]: sure as you instruct i get it

**Comment by dawgdevv** [comment]
@patzielinski inlined it take a look

**Comment by dawgdevv** [comment]
@patzielinski just a heads up on this made the requested improvements

**Review by patzielinski — APPROVED** [review]
